solidity/contracts/isms/ccip-read/AbstractCcipReadIsm.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {ICcipReadIsm} from "../../interfaces/isms/ICcipReadIsm.sol";
+import {IMailbox} from "../../interfaces/IMailbox.sol";
+import {Message} from "../../libs/Message.sol";
+import {AbstractMultisigIsm} from "../multisig/AbstractMultisigIsm.sol";
+
+/// @param sender the address of the contract making the call, usually address(this)
+/// @param urls the URLs to query for offchain data
+/// @param callData context needed for offchain service to service request
+/// @param callbackFunction function selector to call with offchain information
+/// @param extraData additional passthrough information to call callbackFunction with
+error OffchainLookup(
+    address sender,
+    string[] urls,
+    bytes callData,
+    bytes4 callbackFunction,
+    bytes extraData
+);
+
+/**
+ * @title AbstractCcipReadIsm
+ * @notice An ISM that allows arbitrary payloads to be submitted and verified on chain.
+ */
+abstract contract AbstractCcipReadIsm is ICcipReadIsm {
+    // ============ Libraries ============
+
+    using Message for bytes;
+
+    // ============ Constants ============
+
+    // solhint-disable-next-line const-name-snakecase
+    uint8 public constant moduleType =
+        uint8(IInterchainSecurityModule.Types.CCIP_READ);
+
+    // ============ Mutable Storage ============
+
+    /// @notice The URLs to query for offchain data from.
+    string[] public offchainUrls;
+
+    /// @notice Data to pass back to the callback function.
+    // slither-disable-next-line uninitialized-state
+    bytes public offchainExtraData;
```

Comment:
Does this need to be specified here? IMO there is no need to prescribe that this data is stored in storage right?